source/common/router/router.cc

```diff
@@ -1426,7 +1426,7 @@ bool Filter::setupRedirect(const Http::ResponseHeaderMap& headers,
       !callbacks_->decodingBuffer() && // Redirects with body not yet supported.
       location != nullptr &&
       convertRequestHeadersForInternalRedirect(*downstream_headers_, *location) &&
```

Comment:
Looking at calls to setupRedirect, I think that setupRedirect should take headers as r-value reference and be in charge of doing the call to parent_.filter_manager_callbacks_.setResponseHeaders

Reply:
Nevermind, I meant to remove this comment but something in the github UI worked against me.